File: 0001.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_of = {}
        for index, num in enumerate(nums):
            if num not in index_of:
                index_of[num] = []
            index_of[num].append(index)
        for index, num in enumerate(nums):
            second_num = target - num
            if second_num in index_of:
                for second_index in index_of[second_num]:
                    if second_index != index:
                        return [index, second_index]
        return [-1, -1]

File: LeetCode/0001.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, ArrayList<Integer>> values = new HashMap<>();// maps values to lists of indices with those values
        for(int i = 0; i < nums.length; i++) {
            if(!values.containsKey(nums[i])) {
                values.put(nums[i], new ArrayList<Integer>());
            }
            values.get(nums[i]).add(i);
            int currentTarget = target - nums[i];
            if(values.containsKey(currentTarget)) {
                for(int indx: values.get(currentTarget)) {
                    if(indx != i) {
                        return new int[]{indx, i};
                    }
                }
            }
        }
        return new int[]{-1, -1};
    }
}